patch/kconfig-exclusions

```diff
@@ -6,14 +6,20 @@ CONFIG_CGROUP_NET_CLASSID
 CONFIG_NET_CLS_CGROUP
 CONFIG_NETFILTER_XT_MATCH_CGROUP
 CONFIG_CGROUP_NET_PRIO
+###-> mellanox_common-start
+###-> mellanox_common-end
```

Comment:
Are these tags really necessary? This file is (almost) an INI file and should be easy for a python script to parse the sections in this file and append/modify as necessary.

Reply:
I wanted to keep the implementation in sync with kconfig-inclusions. Besides, adding [mellanox], [mellanox-arm64] etc would make the kernel platform specific and was not recommended here https://github.com/sonic-net/sonic-linux-kernel/pull/309#issuecomment-1470134897

Reply:
I meant for both kconfig-exclusions and kconfig-inclusions (and kconfig-force-inclusions, although I need to see if that's still actually needed or not). For instance, the following bit of Python 3 code can parse both files, allows editing the config settings, and then writing them back:

```
>>> import configparser
>>> import sys
>>> configParser = configparser.ConfigParser(allow_no_value=True, strict=False)
>>> configParser.optionxform = str
>>> configParser.read("patch/kconfig-inclusions")
['patch/kconfig-inclusions']
>>> configParser.write(sys.stdout)
[common]
CONFIG_LOG_BUF_SHIFT = 20

[amd64]
CONFIG_I2C_MUX_PCA9541 = m
CONFIG_SENSORS_MAX6697 = m
CONFIG_PCI_REASSIGN_PREF_MEM = y
...
```

`strict=False` is needed because there are currently options specified twice, which should ideally be cleaned up. Fortunately, those duplicates are setting non-conflicting values. Comments aren't preserved though, making the diff size larger than the minimum, but it still functionally works.

I can understand it's needed for the `patch/series` file, because order matters there more so than the kernel configs, but I don't think the kconfig files need the tags.

Reply:
It is certainly doable (ignoring the diff) and is a better approach. it also would need a fix in manage-config (it currently doesn't read platform-arch flags). 

However, it goes against your argument https://github.com/sonic-net/sonic-linux-kernel/pull/309#issuecomment-1470134897 that its preferable to have a platform agnostic kernel. What would you recommend considering that?

Reply:
For amd64, the kernel should still be platform-agnostic. For armhf and arm64, I _think_ it would be a bit difficult to have a platform-agnostic kernel (I could be wrong here; my knowledge of armhf and arm64 is limited compared to amd64), so a platform-specific kernel is fine for now.

For manage-config, I think it currently considers arch, and then platform (separately), right?

Reply:
the script only fetches, common, platform & arch as of today and applies only those. Not [platform-arch] https://github.com/sonic-net/sonic-linux-kernel/blob/1398567e255474c06de6fa93d5ca17ae7d948f6b/manage-config#L143

So, you'd suggestion would be to keep the existing "mellanox-start/mellanox-end" markers (i.e. for amd64) and for arm64/armhf you'd recommend using the ini approach?

Reply:
As discussed online, the amd64 kernel will have same kconfig across all the platforms and so markers are used. 

arm64 can be platform specific and thus a new section [mellanox-arm64] is added.